portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -1,29 +1,43 @@
+const api = require('../../api');
 const sendEmail = require('../../email');
 const { EMAIL_TEMPLATE_IDS } = require('../../../constants');
 
 const { PDC_INPUTTERS_EMAIL_RECIPIENT } = process.env;
 
 const sendEmailToPdcInputtersEmail = async (bankName, month, year) => {
-  await sendEmail(
-    EMAIL_TEMPLATE_IDS.UTILISATION_REPORT_NOTIFICATION,
-    PDC_INPUTTERS_EMAIL_RECIPIENT,
-    {
-      bankName,
-      reportPeriod: `${month} ${year}`,
-    },
-  );
+  await sendEmail(EMAIL_TEMPLATE_IDS.UTILISATION_REPORT_NOTIFICATION, PDC_INPUTTERS_EMAIL_RECIPIENT, {
+    bankName,
+    reportPeriod: `${month} ${year}`,
+  });
 };
 
-const uploadReportAndSendNotification = async (req, res) => {
-  const { bankName, month, year } = req.body;
+const uploadReport = async (req, res) => {
   try {
-    // TODO FN-1103 save file
-    sendEmailToPdcInputtersEmail(bankName, month, year);
-    // TODO FN-1103 change what is sent in the response
-    res.status(200).send(true);
+    const { reportData, month, year, user } = req.body;
+    const parsedReportData = JSON.parse(reportData);
+    const parsedUser = JSON.parse(user);
+
+    // TODO: FN-967 save file to azure
+    // const file = req.file;
+
+    // if (!file) return res.status(400).send();
+    // const path = await saveFileToAzure(req.file, month, year, bank);
+
+    const saveDataResponse = await api.saveUtilisationReport(parsedReportData, month, year, parsedUser, 'a file path');
+
+    if (saveDataResponse.status !== 201) {
+      const status = saveDataResponse.status || 500;
+      console.error('Failed to save utilisation report: %O', saveDataResponse);
+      return res.status(status).send({ status, data: 'Failed to save utilisation report' });
+    }
+
+    sendEmailToPdcInputtersEmail(parsedUser.bank.bankName, month, year);
+
+    return res.status(200).send({ status: 200, data: 'Successfully saved utilisation report' });
```

Comment:
is there any reason to return 200 instead of the original 201 response here?

It doesn't look like we usually include a `status` attribute for success responses. Should we instead return the body of the original success response (i.e. `{ dateUploaded: <some_date_value>}`)